jerry-core/parser/regexp/re-compiler.c

```diff
@@ -624,14 +444,56 @@ re_parse_alternative (re_compiler_ctx_t *re_ctx_p, /**< RegExp compiler context
   return ret_value;
 } /* re_parse_alternative */
 
+static const re_compiled_code_t *re_cache[RE_CHACHE_SIZE];
+
+/**
+ * Search for the given pattern in the RegExp cache
+ *
+ * @return compiled bytecode - if found
+ *         NULL - otherwise
+ */
+static re_compiled_code_t *
+re_find_bytecode_in_cache (ecma_string_t *pattern_str_p, /**< pattern string */
+                           uint32_t *idx) /**< [out] index */
+{
+  uint32_t free_idx = RE_CHACHE_SIZE;
+
+  for (*idx = 0u; *idx < RE_CHACHE_SIZE && re_cache[*idx] != NULL; (*idx)++)
+  {
+    re_compiled_code_t *cached_bytecode_p = re_cache[*idx];
+
+    if ((cached_bytecode_p->flags >> ECMA_BYTECODE_REF_SHIFT) > 0)
+    {
+      ecma_string_t *cached_pattern_str_p;
+      cached_pattern_str_p = ECMA_GET_NON_NULL_POINTER (ecma_string_t, cached_bytecode_p->pattern_cp);
+
+      if (ecma_compare_ecma_strings (cached_pattern_str_p, pattern_str_p))
+      {
+        return re_cache[*idx];
```

Comment:
This is not good. This won't work correctly. The flags should be checked as well here. The first 2 bytes are different on your example.